controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -262,7 +263,11 @@ func (a *Adapter) getSucceededBuildPipelineRunsForComponent(component *applicati
 
 func (a *Adapter) annotateBuildPipelineRunWithSnapshot(pipelineRun *tektonv1beta1.PipelineRun, snapshot *applicationapiv1alpha1.Snapshot) (*tektonv1beta1.PipelineRun, error) {
 	patch := client.MergeFrom(pipelineRun.DeepCopy())
-	h.AddAnnotation(&pipelineRun.ObjectMeta, tekton.SnapshotNameLabel, snapshot.Name)
+	newAnnotations := map[string]string{}
+	newAnnotations[tekton.SnapshotNameLabel] = snapshot.Name
+
+	_ = metadata.AddAnnotations(&pipelineRun.ObjectMeta, newAnnotations)
+
```

Comment:
What is the benefit of this change?

Now I have to write this boilerplate code to add single annotation into list, to update it and copy paste it, instead of having a single functions that hides it

Reply:
Also could be just
```
newAnnotations := map[string]string{
  tekton.SnapshotNameLabel: snapshot.Name
}
```